src/ps3d.f90

```diff
@@ -80,15 +80,26 @@ subroutine pre_run
             ! calculate the initial \xi and \eta mean and save it in ini_vor_mean:
             ini_vor_mean = calc_vorticity_mean()
 
+            call calculate_peref
+
             call vor2vel
 #ifdef ENABLE_BUOYANCY
             bbdif = maxval(buoy) - minval(buoy)
 #else
             bbdif = zero
 #endif
             ke = get_kinetic_energy()
+            pe = get_potential_energy()
+            te = ke + pe
             en = get_enstrophy()
-            call init_diffusion(bbdif, ke, en)
+
+#ifdef ENABLE_BUOYANCY
+            ! add buoyancy term to enstrophy
+            en = en + get_gradb_integral()
```

Comment:
OK, good, here you create the generalised enstrophy - answers one of my comments!